pkg/env/env.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package env
+
+import (
+	"fmt"
+	"github.com/a8m/envsubst"
+	"sort"
+	"strings"
+)
+
+type Environment []Var
+
+// ExpandEnv expands the environments supporting the notation "${var:-$DEFAULT}"
+func ExpandEnv(value string, expandIfEmpty bool) (string, error) {
```

Comment:
NIT: Probably this is one of the functions that is giving us more headache because of the control flag `expandIfEmpty`. What do you think about creating two different methods to make it more explicit?

Reply:
It sounds like a good idea and moment to do it, changed in: c2c2a1b6